src/main/java/net/javadiscord/javabot2/systems/moderation/SpamListener.java

```diff
@@ -0,0 +1,34 @@
+package net.javadiscord.javabot2.systems.moderation;
+
+import net.javadiscord.javabot2.Bot;
+import org.javacord.api.event.message.MessageCreateEvent;
+import org.javacord.api.listener.message.MessageCreateListener;
+
+import java.time.Duration;
+import java.time.Instant;
+
+/**
+ * Listens for spam using the {@link MessageCache}.
+ */
+public class SpamListener implements MessageCreateListener {
```

Comment:
It doesn't really make sense to have a second listener, because now each time a message is sent, two listeners in parallel are going to try and process it for spam. Just add the logic from the MessageCache class to this class.